lfsapi/client.go

```diff
@@ -23,20 +23,37 @@ import (
 const MediaType = "application/vnd.git-lfs+json; charset=utf-8"
 
 var (
-	UserAgent = "git-lfs"
-	httpRE    = regexp.MustCompile(`\Ahttps?://`)
+	UserAgent                 = "git-lfs"
+	httpRE                    = regexp.MustCompile(`\Ahttps?://`)
+	maxSshAuthenticateRetries = 5
 )
 
-func (c *Client) NewRequest(method string, e Endpoint, suffix string, body interface{}) (*http.Request, error) {
-	sshRes, err := c.SSH.Resolve(e, method)
-	if err != nil {
+func (c *Client) sshResolveWithRetries(e Endpoint, method string) (sshAuthResponse, error) {
+	var sshRes sshAuthResponse
+	var err error
+
+	requests := tools.MaxInt(0, maxSshAuthenticateRetries) + 1
+	for i := 0; i < requests; i++ {
+		sshRes, err = c.SSH.Resolve(e, method)
+		if err == nil {
+			return sshRes, nil
+		}
+
 		tracerx.Printf("ssh: %s failed, error: %s, message: %s",
```

Comment:
Should these messages be changed to indicate when we are and aren't retrying? You can take a look at something similar, like this:

https://github.com/git-lfs/git-lfs/blob/8c8a711404391758565a946fdc36e8249344b1f1/tq/transfer_queue.go#L493